src/gsy_e/models/strategy/order_updater.py

```diff
@@ -12,36 +15,89 @@ class OrderUpdaterParameters:
     Parameters of the order updater class. Includes start / end energy rate and time interval
     between each price update.
     """
-    update_interval: duration
-    initial_rate: float
-    final_rate: float
+
+    update_interval: Optional[duration]
+    initial_rate: Optional[Union[dict[DateTime, float], float]]
+    final_rate: Optional[Union[dict[DateTime, float], float]]
 
     def serialize(self):
         """Serialize parameters."""
         return {
-            "update_interval": self.update_interval,
-            "initial_rate": self.initial_rate,
-            "final_rate": self.final_rate
+            "update_interval": self.get_update_interval(),
+            "initial_rate": (
+                self.initial_rate
+                if isinstance(self.initial_rate, (type(None), float, int))
+                else convert_pendulum_to_str_in_dict(self.initial_rate)
+            ),
+            "final_rate": (
+                self.final_rate
+                if isinstance(self.final_rate, (type(None), float, int))
+                else convert_pendulum_to_str_in_dict(self.final_rate)
+            ),
         }
 
+    def get_update_interval(self):
+        """Return update_interval. If not set, return default value."""
+        if self.update_interval is None:
+            return duration(minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)
+        return self.update_interval
+
+    @staticmethod
+    def _get_default_value_initial_rate(_time_slot: DateTime):
```

Comment:
Hmmm, it feels we can simplify here no and just return 0 instead of calling this function? Or the idea is to have a default value for every time_slot?

Reply:
Please have a look at the `HeatPumpOrderUpdaterParameters` that inherits from the `OrderUpdaterParameters`. This is to inject a different default value to every implementation of `OrderUpdaterParameters`. Here we do not need a specific default value. Does it make sense?

Reply:
I see we are overriding the function. Looks fine then